packages/components/src/navigation/stories/index.js

```diff
@@ -132,3 +136,73 @@ function Example() {
 export const _default = () => {
 	return <Example />;
 };
+
+function ComponentsCompositionExample() {
+	const [ activeItem, setActiveItem ] = useState( 'item-1' );
+	return (
+		<Navigation2>
+			{ ( { activeLevel, navigateTo, NavigationBackButton } ) => (
+				<>
+					<Navigation2Level
+						slug="root"
+						title="Home"
+						activeLevel={ activeLevel }
+					>
+						<Navigation2Item
+							slug="item-1"
+							title="Item 1"
+							onClick={ () => {
+								navigateTo( 'sub-menu' );
+							} }
+							activeItem={ activeItem }
+						/>
+					</Navigation2Level>
+
+					<Navigation2Level
+						slug="sub-menu"
+						title="Sub Menu"
+						activeLevel={ activeLevel }
+						navigateBack={
+							<NavigationBackButton
+								parentLevel="root"
+								label="Back to Home"
+							/>
+						}
+					>
+						<Navigation2Item
+							slug="item-2"
+							title="Item 2"
+							onClick={ () => {
+								navigateTo( 'nested-sub-menu' );
+							} }
+							activeItem={ activeItem }
+						/>
+					</Navigation2Level>
+
+					<Navigation2Level
+						slug="nested-sub-menu"
+						title="SubSub Menu"
+						activeLevel={ activeLevel }
+						navigateBack={
+							<NavigationBackButton
+								parentLevel="sub-menu"
+								label="Back to Sub Menu"
+							/>
+						}
+					>
+						<Navigation2Item
+							slug="item-3"
+							title="Item 3"
+							onClick={ () => {
+								setActiveItem( 'item-3' );
+							} }
```

Comment:
I'm intrigued by the idea of `navigateTo`. In this example, its used to move the Navigation over, but not actually set the activeItem. I think we could do this internally if the item has children and change `onClick` to `onSelect`.

Reply:
I kept navigation and selection separate as they are separate concerns imho.
My reasoning was that I don't see the use of activating the parent item, when navigating to a sub menu will hide the item.
I'm probably missing some context though. 🤔